src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -1064,6 +1076,60 @@ public static String slurp(final InputStream is, final Charset charSet) {
     }
 
     /**
+     * Go through the files provided and if they have one of the provided file extensions pass the file into the output
+     * otherwise assume that file is a list of filenames and unfold it into the output.
```

Comment:
This method also recursively expands files of filenames, if one of the filenames in a file of filenames points to another file of filenames. If this is the intended behavior it should be documented and tested.

Reply:
is tested, will update message.